docker-app/qfieldcloud/core/tests/test_queryset.py

```diff
@@ -310,3 +310,63 @@ def p(proj, user):
         self.assertEqual(p(self.project9, self.user3).project_role, ProjectCollaborator.Roles.EDITOR)
         self.assertEqual(p(self.project9, self.user3).project_role_origin, ProjectQueryset.RoleOrigins.TEAMMEMBER.value)
         # fmt: on
+
+    def test_private_project_memberships(self):
+        """Tests for QF-1553 - limit collaboration on private projects"""
+
+        def assert_no_role(p, u):
+            """Asserts that user has no role on project"""
+            # Test on Users
+            with self.assertRaises(User.DoesNotExist):
+                User.objects.for_project(p).get(pk=u.pk)
+
+            # Test on Project
+            with self.assertRaises(Project.DoesNotExist):
+                Project.objects.for_user(u).get(pk=p.pk)
+
+        def assert_role(p, u, expected_role, expected_origin):
+            """Asserts that user has give role/origin on project"""
+            # Test on Users
+            user = User.objects.for_project(p).get(pk=u.pk)
+            self.assertEqual(user.project_role, expected_role)
+            self.assertEqual(user.project_role_origin, expected_origin.value)
+
+            # Test on Project
+            project = Project.objects.for_user(u).get(pk=p.pk)
+            self.assertEqual(project.user_role, expected_role)
+            self.assertEqual(project.user_role_origin, expected_origin.value)
+
+        # Shorthands
+        roles = ProjectCollaborator.Roles
+        origins = ProjectQueryset.RoleOrigins
+
+        # Initial user setup
+        u = User.objects.create(username="u")
+        o = Organization.objects.create(username="o", organization_owner=u)
+        p = Project.objects.create(name="p", owner=u, is_public=True)
+
+        u1 = User.objects.create(username="u1")
+
+        # A public project is readable
+        # TODO: this fails, is is expected the queryset does not contain public projects ?
+        # assert_role(p, u1, roles.READER, origins.PUBLIC)
```

Comment:
The docstring of `User.objects.for_projects` actually states that `If the project is public, it will return only the directly collaborated users.`, so this is fine

I expected `User.objects.for_projects` to by symmetric with `Project.objects.for_user`